test/creditguild-test.js

```diff
@@ -0,0 +1,206 @@
+      result2 = resp2.events[resp2.events.length - 1].args.result;
+      expect(result2).to.equal(false);
+
+    });
+  });
+
+  describe("Register", function () {
+    it("Should register an array of members", async function () {
+
+      // unhappy path - should revert because ALICE is not a member
+      await expect(creditGuild.register(
+        [MEMBER1.address, MEMBER2.address, ALICE.address]
+      )).to.be.revertedWith('!member');
+
+      // unhappy path - all 3 of the members are not vouching for msg.sender
+      // TODO: with the mock set as always vouching 1, we are unable to test this
+
+      // happy path
+      const addressToRegister = ALICE.address;
+
+      const tx = await creditGuild.checkIsMember(addressToRegister);
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.result;
+      expect(result).to.equal(false);
+
+      await creditGuild.connect(ALICE).register([MEMBER1.address, MEMBER2.address, MEMBER3.address]);
+
+      const tx1 = await creditGuild.checkIsMember(addressToRegister);
+      const resp1 = await tx1.wait();
+      result1 = resp1.events[resp1.events.length - 1].args.result;
+      expect(result1).to.equal(true);
+
+      // unhappy path - should revert because msg.sender is already a member
+      await expect(creditGuild.connect(ALICE).register(
+        [MEMBER1.address, MEMBER2.address, MEMBER3.address]
+      )).to.be.revertedWith('member');;
+
+    });
+  });
+
+  describe("Claim Vouch", function () {
+    it("Should allow an address to claim a vouch", async function () {
+
+      // get current vouch amount
+      vouchAmount = await creditGuild.vouchAmount();
+
+      tx = await creditGuild.claimVouch();
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.vouchAmount;
+      expect(result).to.equal(vouchAmount);
+
+    });
+  });
+
+  describe("Burn Membership", function () {
+    it("Should burn a membership to the DAO", async function () {
+
+      // happy path
+      tx = await creditGuild.checkIsMember(MEMBER1.address);
+      const resp = await tx.wait();
+      result = resp.events[resp.events.length - 1].args.result;
+      expect(result).to.equal(true);
+
+      tx2 = await creditGuild.burnMembership(MEMBER1.address);
+      const resp2 = await tx2.wait();
+      result2 = resp2.events[resp2.events.length - 1].args.member;
+      expect(result2).to.equal(MEMBER1.address);
+
+      tx3 = await creditGuild.checkIsMember(MEMBER1.address);
+      const resp3 = await tx3.wait();
+      result3 = resp3.events[resp3.events.length - 1].args.result;
+      expect(result3).to.equal(false);
+
+      // unhappy path
+      tx4 = await creditGuild.checkIsMember(JACK.address);
+      const resp4 = await tx4.wait();
+      result4 = resp4.events[resp4.events.length - 1].args.result;
+      expect(result4).to.equal(false);
+
+      await expect(creditGuild.burnMembership(JACK.address)).to.be.revertedWith('!member');
```

Comment:
I would try and seperate out unhappy paths into a seperate it function ie:


```js
describe("@method burnMembership()", () => {
  it("happy path: removes member from the DAO", async () => {
     // ....
  });

  it("happy path: burns the members NFT", async () => {
     // ....
  });

  it("unhappy path: cannot burnMembership for non member", async () => {
     // ....
  });
});
```

Reply:
In this you also want to include a test to check that the NFT has been burnt something like:

```js
const bal = await creditGuild.balanceOf(member);
expect(bal).lte(0);
```